app/code/Magento/Search/Test/Mftf/Test/StorefrontVerifySearchFieldVisibilityWhenSuggestionDisabledTest.xml

```diff
@@ -0,0 +1,44 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+ /**
+  * Copyright © Magento, Inc. All rights reserved.
+  * See COPYING.txt for license details.
+  */
+-->
+
+<tests xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+       xsi:noNamespaceSchemaLocation="urn:magento:mftf:Test/etc/testSchema.xsd">
+    <test name="StorefrontVerifySearchFieldVisibilityWhenSuggestionDisabledTest">
+        <annotations>
+            <stories value="Search Term"/>
+            <title value="Mini search field appears if suggestions was disabled"/>
+            <description value="Mini search field appears if suggestions was disabled"/>
+            <severity value="CRITICAL"/>
+            <testCaseId value="MC-38652"/>
+            <group value="mtf_migrated"/>
+        </annotations>
+
+        <before>
+            <!-- Login as admin -->
+            <actionGroup ref="AdminLoginActionGroup" stepKey="loginAsAdmin"/>
+            <!-- Disable search suggestion -->
+            <magentoCLI command="config:set catalog/search/search_suggestion_enabled 0" stepKey="disableSearchSuggestion"/>
```

Comment:
Hi @mrtuvn 
To persist the test isolation, we need to rollback all the changes (side-effects) introduced in the test. So, set the search suggestions back to "enabled" and logout from the admin panel. A common practice is to do it in the `<after>` section. 

Thanks.

Reply:
Updated as your suggest